src/kbc_modules/cc_kbc/crypto/mod.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use aes_gcm::aead::{Aead, NewAead};
+use aes_gcm::Aes256Gcm;
+use anyhow::*;
+use rsa::pkcs8::EncodePublicKey;
+use rsa::{PaddingScheme, RsaPrivateKey, RsaPublicKey};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha384};
+
+const RSA_ALGORITHM: &str = "rsa-pkcs1v15";
+const RSA_PUBKEY_LENGTH: usize = 2048;
+
+const AES_GCM_256_ALGORITHM: &str = "aes-gcm-256";
+
+// The key inside TEE to decrypt confidential datas.
+#[derive(Debug, Clone)]
+pub struct TeeKey {
+    private_key: RsaPrivateKey,
+    public_key: RsaPublicKey,
+}
+
+impl TeeKey {
+    pub fn new() -> Result<TeeKey> {
+        let mut rng = rand::thread_rng();
+
+        let private_key = RsaPrivateKey::new(&mut rng, RSA_PUBKEY_LENGTH)?;
+        let public_key = RsaPublicKey::from(&private_key);
+
+        Ok(TeeKey {
+            private_key,
+            public_key,
+        })
+    }
+
+    // Export TEE public key as specific structure.
+    pub fn export_pubkey(&self) -> Result<TeePubKey> {
+        let pem_line_ending = rsa::pkcs1::LineEnding::default();
+        let pubkey_pem_string = self.public_key.to_public_key_pem(pem_line_ending)?;
+
+        Ok(TeePubKey {
+            algorithm: RSA_ALGORITHM.to_string(),
+            pubkey_length: RSA_PUBKEY_LENGTH.to_string(),
+            pubkey: pubkey_pem_string,
+        })
+    }
+
+    // Use TEE private key to decrypt cipher text.
+    pub fn decrypt(&self, cipher_text: Vec<u8>) -> Result<Vec<u8>> {
+        let padding = PaddingScheme::new_pkcs1v15_encrypt();
```

Comment:
This can not be a const, but can only be a lazy static. But I don't think it is necessary to use it as a global static variable, because it will only be used here.

Reply:
Yes. I wanted to keep the RSA_ALGORITHM and the generation of padding close together in the source code (I'm worried for a future change in which we change one but not the other). We can do it using something like:

```rust
const RSA_ALGORITHM: &str = "rsa-pkcs1v15";
const NEW_PADDING: fn() -> PaddingScheme = PaddingScheme::new_pkcs1v15_encrypt;
```

and then in `decrypt()` use:

```rust
let padding = NEW_PADDING();
```

but that's a bit of over-engineering for this.

Maybe some other way of keeping them together? Maybe a CryptoType struct with `name` (string) and `new_padding()` function? Not sure it's worth it.